### PR TITLE
Feature/ASAA-20 Commit Screen Branch Info Additions

### PR DESCRIPTION
**Pull request by LuisRocketRangel**
ASAA-20 Add Branch Info to Commit Screen 

Requirements:
-ADd Branch Info to Commit Screen UI

Solution:
-Modify API call to get branch info as well as commit info
-Add UI element to display branch name
-Add branch name list to the commit screen state

### CODE REVIEW

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt [hunk 1]
```diff
@@ -47,27 +47,43 @@ class RepositoryCommitViewModel : BaseViewModel() {
...
+        val commitList = mutableListOf<CommitDto>()
         path.setValue(name)
         selectedRepo?.let {
             val slug = it.workspaceDto?.slug ?: ""
             val repoName = it.name ?: ""
             launchIO {
-                val result = repo.getCommits(slug, repoName)
-                when (result) {
-                    is Status.Success -> srcCommits.value = result.data
+                when (val branchCallResult = repo.getBranches(slug, repoName)) {
```
Comment (BottleRocket-Colin) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt]: Look into using handlingErrors extension function here.

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryCommitScreen.kt [hunk 2]
```diff
@@ -79,26 +79,29 @@ fun CommitItem(item: RepositoryCommitItemUiModel) {
                     style = MaterialTheme.typography.h3.copy(color = Colors.tertiary)
                 )
                 Text(
-                    text = item.hash,
-                    style = MaterialTheme.typography.h5
+                    text = item.branchName,
+                    style = MaterialTheme.typography.h5.copy(fontWeight = FontWeight.Bold),
```
Comment (BottleRocket-Colin) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryCommitScreen.kt]: We have a toBold() extension function that performs this copy.

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryCommitScreen.kt [hunk 3]
```diff
@@ -79,26 +79,29 @@ fun CommitItem(item: RepositoryCommitItemUiModel) {
...
-                    modifier = Modifier.padding(
-                        top = Dimens.grid_2,
-                        end = Dimens.grid_2
-                    ),
                     text = item.timeSinceCommitted.getString(),
                     style = MaterialTheme.typography.h5
                 )
+                Text(
+                    text = item.hash,
+                    style = MaterialTheme.typography.h5.copy(fontWeight = FontWeight.Bold)
```
Comment (BottleRocket-Colin) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryCommitScreen.kt]: ditto w/ respect to toBold()

**Review by BottleRocket-Colin — CHANGES_REQUESTED** [review]
A couple of minor changes,

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt (hunk 1 above)
Comment (LuisRocketRangel) [reply app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt]: Donzo!

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitStateConverter.kt [hunk 4]
```diff
@@ -8,5 +8,6 @@ import com.bottlerocketstudios.compose.repository.RepositoryCommitScreenState
 fun RepositoryCommitViewModel.toState() = RepositoryCommitScreenState(
     path = path.collectAsState(),
     itemCount = itemCount.collectAsState(),
-    commitItems = uiModels.collectAsState()
+    commitItems = uiModels.collectAsState(),
+    branchNames = branchItems.collectAsState()
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitStateConverter.kt:12]: @LuisRocketRangel nit: Let's add a trailing comma here to allow for simpler diffs if/when new arguments are added to `RepositoryCommitScreenState` in the future

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt [hunk 5]
```diff
@@ -47,27 +46,36 @@ class RepositoryCommitViewModel : BaseViewModel() {
                     author = commit.author?.userInfo?.displayName ?: "",
                     timeSinceCommitted = commit.date.formattedUpdateTime(clock),
                     hash = commit.hash?.take(HASH_LENGTH) ?: "",
-                    message = commit.message ?: ""
+                    message = commit.message ?: "",
+                    branchName = commit.branchName ?: ""
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt]: @LuisRocketRangel nit: Trailing comma :)

File: data/src/main/java/com/bottlerocketstudios/brarchitecture/data/model/CommitDto.kt [hunk 6]
```diff
@@ -15,5 +15,6 @@ data class CommitDto(
     @Json(name = "type") val type: String?,
     @Json(name = "hash") val hash: String?,
     @Json(name = "author") val author: AuthorDto?,
-    @Json(name = "repository") val commitRepository: CommitRepositoryDto?
+    @Json(name = "repository") val commitRepository: CommitRepositoryDto?,
+    var branchName: String?
```
Comment (jaredstockton) [inline data/src/main/java/com/bottlerocketstudios/brarchitecture/data/model/CommitDto.kt]: @LuisRocketRangel We should avoid adding additional data to Dto classes, even moreso mutable data. Going to work on an alternate solution and push up a review commit later today

**Review by jaredstockton — CHANGES_REQUESTED** [review]
@LuisRocketRangel Working on a review commit with some additional changes here

**Review by BottleRocket-Colin — APPROVED** [review]

**Review by jaredstockton — APPROVED** [review]

**Pull request merged**